src/main/java/filter/expression/Qualifier.java

```diff
@@ -136,23 +131,61 @@ public static boolean process(IModel model, FilterExpression expr, TurboIssue is
         return exprWithNormalQualifiers.isSatisfiedBy(model, issue, new MetaQualifierInfo(metaQualifiers));
     }
 
+    /**
+     * Get all milestones that can be aliased with the keyword "current".
+     */
+    private static List<TurboMilestone> getMilestonesWithAlias(IModel model, List<String> repoIds) {
+        List<TurboMilestone> milestones = new ArrayList<>();
+        // add milestones with due date first
+        milestones.addAll(model.getMilestones().stream()
+                .filter(ms -> ms.getDueDate().isPresent())
+                .filter(ms -> repoIds.contains(ms.getRepoId().toLowerCase()))
+                .sorted((a, b) -> getMilestoneDueDateComparator().compare(a, b))
+                .collect(Collectors.toList()));
+
+        // Special case: if there is only one open milestone, it should be included regardless of whether
+        // it has due date or not.
+        // In this case, if it is not included already (the open milestone does not
+        // have due date), append it to the milestone list.
+        List<TurboMilestone> openMilestones = getOpenMilestones(model, repoIds);
+        if (openMilestones.size() == 1) {
+            TurboMilestone openMilestone = openMilestones.get(0);
+            boolean hasDueDate = openMilestone.getDueDate().isPresent();
+            if (!hasDueDate) {
+                milestones.add(openMilestone);
+            }
+        }
+
+        return milestones;
+    }
+
+    private static List<TurboMilestone> getOpenMilestones(IModel model, List<String> repoIds) {
+        return model.getMilestones().stream()
+                .filter(ms -> ms.isOpen())
+                .filter(ms -> repoIds.contains(ms.getRepoId().toLowerCase()))
+                .collect(Collectors.toList());
+    }
+
     private static Optional<Integer> getCurrentMilestoneIndex(List<TurboMilestone> allMilestones) {
```

Comment:
The logic can be like this?
1. Sort by due date
2. traverse the list until the the best match for 'current' is found
3. traverse the list again to add milestones to the map.

So, step 2 is where we locate the current milestone. Yes, if it cannot be done in this method without breaking SLAP, it may have to be extracted as a method.